src/main/java/uk/gov/justice/hmpps/datacompliance/jobs/imageupload/OffenderIterator.java

```diff
@@ -95,13 +103,23 @@ private OffenderNumbersResponse applyForBatch(final OffenderAction action, final
 
             currentThread().interrupt();
 
-            throw new IllegalStateException("Execution of batch interrupted", e);
+            throw new IllegalStateException("Execution interrupted", e);
         }
 
         return response;
     }
 
-    private long indexOfFinalBatch(final OffenderNumbersResponse response) {
+    private OffenderNumbersResponse getOffenderNumbers(final long offset,
+                                                       final long limit,
+                                                       final ImageUploadBatch batch) {
+
+        return repository.findFirstByBatchIdNotOrderByUploadStartDateTimeDesc(batch.getBatchId())
+                .map(lastUpload -> prisonApiClient.getOffendersWithNewImages(
+                        lastUpload.getUploadStartDateTime().toLocalDate(), offset, limit))
+                .orElseGet(() -> prisonApiClient.getOffenderNumbers(offset, limit));
+    }
```

Comment:
Find the last (not current) upload batch and if it exists, only request offenders with new images captured from that date onwards